Address UX feedback tweaks.

prodonjs opened the pull request:
- Shortens sidebar links and thins divider to 1px.
- Adds secondary text to platform links
- Fixes hover / active state for dashboard card links.
- Address a11y issue for tabbing on iframe-links.
- Fixes an invalid selector in main-page unit tests.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/kubeflow/3356)
<!-- Reviewable:end -->

prodonjs commented:
/cc @avdaredevil 
/cc @ajayalfred 
/area front-end
/area centraldashboard

This addresses the majority of the tweaks we talked about yesterday afternoon save for the font-weight: 500 issue.

I didn't deploy it in cluster, but I recorded this [screencast](https://screencast.googleplex.com/cast/NTk0NTQ2NTA0NjU2NDg2NHwxOTg1YjQyNS1hMg) showing the correct hover and active behaviors along with the adding secondary text lines for the platform cards.

avdaredevil commented on components/centraldashboard/public/components/dashboard-view.css:
```diff
@@ -40,32 +55,39 @@ paper-card {
         flex: 1;
-    };
+    }
+    ;
```
This is really weird formatting...

avdaredevil commented on components/centraldashboard/public/components/dashboard-view.css:
```diff
@@ -15,15 +17,28 @@
     overflow: hidden;
-    min-width: 23em; /* 320px */
+    min-width: 23em;
+    /* 320px */
```
This is really weird formatting

avdaredevil commented on components/centraldashboard/public/components/main-page_test.js:
```diff
@@ -124,8 +124,8 @@ describe('Main Page', () => {
-        const sidebarLinkSelector = '#MainDrawer iron-selector a.iframe-link';
-        const headerLinkSelector = 'app-header paper-tabs a';
+        const sidebarLinkSelector = '#MainDrawer iron-selector iframe-link';
+        const headerLinkSelector = '#ViewTabs paper-tabs a';
```
Nice! 💯

prodonjs commented:
Latest commit adds the font-weight support which looks like this.

![Screenshot from 2019-05-24 12-25-16](https://user-images.githubusercontent.com/6835846/58342722-29177580-7e1f-11e9-8e35-c5f7802465f7.png)

I'll try to clean up the automatic CSS formatting that my IDE is using. I don't think we have a standard template that's enforced anywhere so mine is probably using whatever the Google default is.

avdaredevil commented:
@prodonjs looks great, however regarding your point about style guide.
Mix-ins in CSS are not part of standard CSS, therefore your processor is
Mis-formatting.

On Fri, May 24, 2019, 12:28 PM Jason Prodonovich <notifications@github.com>
wrote:

> Latest commit adds the font-weight support which looks like this.
>
> [image: Screenshot from 2019-05-24 12-25-16]
> <https://user-images.githubusercontent.com/6835846/58342722-29177580-7e1f-11e9-8e35-c5f7802465f7.png>
>
> I'll try to clean up the automatic CSS formatting that my IDE is using. I
> don't think we have a standard template that's enforced anywhere so mine is
> probably using whatever the Google default is.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubeflow/kubeflow/pull/3356?email_source=notifications&email_token=ABIOV2XDQ75X7U35SPSAYHDPXAJSHA5CNFSM4HPQ2YW2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWF4FCA#issuecomment-495698568>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABIOV2UQLFTYTWTV5HNKGUTPXAJSHANCNFSM4HPQ2YWQ>
> .
>

ajayalfred commented:
Hover states look fantastic! Thanks Jason. Glad to see font-weight: 500 can be loaded. Card headers look much better.

Quick note on the left nav font-styles

**Active state CSS (nav item currently selected):**
font-family: GoogleSans;
font-weight: 500;
font-size: 15px;
color: #FFFFFF;
letter-spacing: 0.3px;
line-height: 24px;

**Inactive state CSS (all other nav items):**
opacity: 0.5;
font-family: GoogleSans;
font-weight: 500;
font-size: 15px;
color: #FFFFFF;
letter-spacing: 0.3px;
line-height: 24px;

ajayalfred commented:
Change 'Overview' to 'Home' in nav

prodonjs commented:
@ajayalfred Thanks for the feedback. I addressed those comments and pushed the image to https://kubeflow-1.endpoints.prodonjs-kubeflow-dev.cloud.goog/. PTAL when you get the chance.

@avdaredevil I figured it was related to the Polymer-specific mix-ins because I just use the standard beautify formatting plugin. It'd be great if there was a way to have a formatter that respects the mix-in syntax, but for now I just disabled formatting on save within this project.

ajayalfred commented:
Looks fantastic!! Really digging the changes with Google sans weight: 500 and the nav font changes!!

One final cosmetic improvement. I've noticed that the contrast between the gray canvas and the light-blue container for the GCP card isn't sufficient. They kind of blend into each other. See below.

![image](https://user-images.githubusercontent.com/2425179/58497532-004df380-8131-11e9-9fc0-f8053e06fb7c.png)

I think it might be best to use the standard card header treatment but with the GCP logo and blue color. See below.
![image](https://user-images.githubusercontent.com/2425179/58497474-e44a5200-8130-11e9-892a-a405f8fa5564.png)

Hope this is an easy fix!

avdaredevil commented:
This should just be a one-line change. I think @prodonjs already tackled some of my AI from the UX meeting, so this can be part of it.

prodonjs commented:
Latest change removes the background and also makes the line-height for the GCP links consistent with the quick shortcuts. 

**Before**
![Screenshot from 2019-05-28 13-24-47](https://user-images.githubusercontent.com/6835846/58498413-fdacc780-814b-11e9-8d98-340b1ec32bb8.png)


**After**
![rffzXGot2i9](https://user-images.githubusercontent.com/6835846/58498359-e79f0700-814b-11e9-9797-9395ed0d728f.png)

ajayalfred commented:
Perfect! thanks!

prodonjs commented:
@avdaredevil Latest commit adds the aria-role we discussed this morning. Let me know if there's anything else.

avdaredevil commented:
/approve
/lgtm
